tests/generator/CodeBlockTests.cs

```diff
@@ -0,0 +1,150 @@
+using System.Collections.Generic;
+using System.IO;
+using NUnit.Framework;
+
+namespace bgen_tests {
+	public class CodeBlockTests {
+		string PerformWriting (ICodeBlock codeBlock)
+		{
+			MemoryStream memoryStream = new ();
+			StreamWriter writer = new StreamWriter (memoryStream);
+			codeBlock.Print (writer);
+			writer.Flush ();
+			memoryStream.Position = 0;
+			using StreamReader reader = new StreamReader (memoryStream);
+			return reader.ReadToEnd ();
+		}
+
+		[Test]
+		public void LineBlockTest ()
+		{
+			string inputText1 = "using System;";
+			string inputText2 = "using Network;";
+			string expectedText = "{\n    using System;\n    using Network;\n}\n";
+			CodeBlock codeBlock = new ();
+			codeBlock.AddLine (inputText1);
+			codeBlock.AddLine (inputText2);
```

Comment:
Compared to something like this:

```cs
print ("using System;");
print ("using Network;");
```

this code is both over twice as long and harder to read.

Something like this would be both shorter and easier to read:

```cs
var codeBlock = new CodeBlock (
	"using System;",
	"using System;",
};
```

which could be done if `CodeBlock` had a constructor like this:

```cs
public CodeBlock (params string[] lines) { ... }
```